Document/0x06g-Testing-Network-Communication.md

```diff
@@ -222,7 +222,9 @@ Verify that the server certificate is pinned. Pinning can be implemented on vari
 2. Limiting certificate issuer to e.g. one entity and bundling the intermediate CA's public key into the application. In this way we limit the attack surface and have a valid certificate.
 3. Owning and managing your own PKI. The application would contain the intermediate CA's public key. This avoids updating the application every time you change the certificate on the server, due to e.g. expiration. Note that using your own CA would cause the certificate to be self-singed.
 
-The code presented below shows how it is possible to check if the certificate provided by the server matches the certificate stored in the app. The method below implements the connection authentication and tells the delegate that the connection will send a request for an authentication challenge.
+The latest approach recommended by Apple is to specify a pinned CA public key in the `Info.plist` file under App Transport Security Settings. You can find an example in their article [Identity Pinning: How to configure server certificates for your app](https://developer.apple.com/news/?id=g9ejcf8y "Identity Pinning: How to configure server certificates for your app").
+
+Another common approach is to check if the certificate provided by the server matches the certificate stored in the app. The method below implements the connection authentication and tells the delegate that the connection will send a request for an authentication challenge.
 
 The delegate must implement `connection:canAuthenticateAgainstProtectionSpace:` and `connection: forAuthenticationChallenge`. Within `connection: forAuthenticationChallenge`, the delegate must call `SecTrustEvaluate` to perform customary X.509 checks. The snippet below implements a check of the certificate.
```

Comment:
do you think that we still need the code below? what if we remove it since we also say "Note that the certificate pinning example above has a major drawback when you use certificate pinning and the certificate changes".

The article that you're referencing has info regarding this. If we link to thst section in the article it should be sufficient.
We're trying to make the MSTG more maintainable, I think this is a good chance to address that. It would be great if you can help us with that. One approach is by identifying things like this that can be found in official sources and simply linking to them instead of maintain explanations and code. Thanks a lot in advance!

Reply:
Yeah, I agree. I've heard that MSTG is moving away from using code examples because they are not easy to maintain, especially when there are official sources with similar examples.
I'm going to update it right now.

Reply:
Also, I think that we still need to mention this approach (without the detailed code example) because it is one of the ways pinning can be implemented. This is a static analysis section, so it looks useful to me to mention different ways of implementation.